### PR TITLE
[ONEM-22712] LISA: maintenance and cleanup implementation

### PR DESCRIPTION
**Pull request by szymon-polke-red**

### CODE REVIEW

**Review by piotr-marcinkowski-red — APPROVED** [review]

File: LISA/Executor.cpp [hunk 1]
```diff
@@ -51,6 +51,74 @@ std::string generateHandle()
...
+    std::string id;
+    std::string version;
+};
+
+std::ostream& operator<<(std::ostream& out, const AppId& app)
+{
+    return out << "app[" << app.type << ":" << app.id << ":" << app.version << "]";
+}
+
+std::vector<AppId> scanDirectories(const std::string& appsPath)
```
[the remark this answers is not in the document]
Comment (szymon-polke-red) [reply LISA/Executor.cpp:71]: As agreed we'll leave this name for now at least.

File: LISA/Executor.cpp [hunk 2]
```diff
@@ -471,11 +539,79 @@ void Executor::doUninstall(std::string type, std::string id, std::string version
...
+        // clear tmp
+        Filesystem::removeDirectory(config.getAppsTmpPath());
+        Filesystem::createDirectory(config.getAppsTmpPath());
+
+        // remove installed apps data not present in installed_apps
+        auto appsPathRoot = config.getAppsPath() + Filesystem::LISA_EPOCH + '/';
+        auto foundApps = scanDirectories(appsPathRoot);
+        for (const auto& app : foundApps) {
+            INFO(app);
+            if (dataBase->IsAppInstalled(app.type, app.id, app.version)) {
```
Comment (sverkoye) [inline LISA/Executor.cpp:559]: The logic should be reversed here: if NOT IsAppInstalled then cleanup dirs

A simple install test shows that the app dirs get deleted immediately after installing a DAC app...

File: LISA/Executor.cpp [hunk 3]
```diff
@@ -471,11 +539,79 @@ void Executor::doUninstall(std::string type, std::string id, std::string version
...
+                Filesystem::removeDirectory(path);
+            }
+        }
+
+        // remove apps data not present in apps
+        auto appsStoragePathRoot = config.getAppsStoragePath() + Filesystem::LISA_EPOCH + '/';
+        auto foundAppsStorages = scanDirectories(appsStoragePathRoot);
+        for (const auto& app : foundAppsStorages) {
+            INFO(app);
+            if (dataBase->IsAppData(app.type, app.id)) {
```
Comment (sverkoye) [inline LISA/Executor.cpp:571]: Same here: reverse the logic. if NOT xxx then cleanup

File: LISA/Executor.cpp [hunk 4]
```diff
@@ -471,11 +539,79 @@ void Executor::doUninstall(std::string type, std::string id, std::string version
...
+            }
+
+            auto dataPaths = dataBase->GetDataPaths(details.type, details.id);
+
+            INFO("PATHS DATA:");
+            for (const auto& path : dataPaths) {
+                INFO("path: ", path);
+                auto dataPath = config.getAppsStoragePath() + path;
+                INFO("abs path: ", dataPath);
+                if (Filesystem::directoryExists(dataPath)) {
```
Comment (sverkoye) [inline LISA/Executor.cpp:603]: I expect you also want to reverse the logic here: if NOT directory exists then create it?

**Pull request merged**